### PR TITLE
Small code optimization in protocol.go

### LINKED ISSUE #2702: Use Mock in action/protocol/execution/protocol_test.go
**What would you like to be added**:

**Why is this needed**:

### PR DESCRIPTION
**Pull request by Liuhaai**
Small code optimization when solving issue #2702.
I think we can close #2702, because it is hard to mock blockchain in the protocol_test.go

### CODE REVIEW

**Review by raullenchai — APPROVED** [review]

File: action/protocol/execution/protocol.go [hunk 1]
```diff
@@ -75,15 +75,12 @@ func (p *Protocol) Handle(ctx context.Context, act action.Action, sm protocol.St
...
-	if !ok {
-		return nil
-	}
-	// Reject oversize execution
-	if exec.TotalSize() > ExecutionSizeLimit {
-		return errors.Wrap(action.ErrActPool, "oversized data")
+	if exec, ok := act.(*action.Execution); ok {
+		if exec.TotalSize() > ExecutionSizeLimit {
+			return errors.Wrap(action.ErrActPool, "oversized data")
+		}
```
Comment (dustinxie) [inline action/protocol/execution/protocol.go:83]: let's use the current code `if !ok { return nil }`, this is an early exit style used very commonly in our code

think about if future we need to add another validation check, like gas amount, the early exit style is better than changing to a big if {} statement

File: action/protocol/execution/protocol.go [hunk 2]
```diff
@@ -75,15 +75,12 @@ func (p *Protocol) Handle(ctx context.Context, act action.Action, sm protocol.St
 	return receipt, nil
 }
 
-// Validate validates an execution
+// Validate validates the size of an execution
```
Comment (dustinxie) [inline action/protocol/execution/protocol.go:78]: revert this, current code only checks size, but `Validate` is meant to validate entire action

File: action/protocol/execution/protocol_test.go [hunk 3]
```diff
@@ -521,14 +521,11 @@ func TestProtocol_Validate(t *testing.T) {
...
-	t.Run("Oversized data", func(t *testing.T) {
-		tmpPayload := [32769]byte{}
-		data := tmpPayload[:]
-		ex, err := action.NewExecution("2", uint64(1), big.NewInt(0), uint64(0), big.NewInt(0), data)
-		require.NoError(err)
-		require.Equal(action.ErrActPool, errors.Cause(p.Validate(context.Background(), ex, nil)))
-	})
+	ex, err := action.NewExecution("2", uint64(1), big.NewInt(0), uint64(0), big.NewInt(0), data)
+	require.NoError(err)
+	require.Equal(action.ErrActPool, errors.Cause(p.Validate(context.Background(), ex, nil)))
```
Comment (dustinxie) [inline action/protocol/execution/protocol_test.go:528]: this is good clean-up

**Review by dustinxie — APPROVED** [review]

File: action/protocol/execution/protocol.go (hunk 1 above)
Comment (Liuhaai) [reply action/protocol/execution/protocol.go:83]: good suggestions

**Pull request merged**